setup.py

```diff
@@ -1,107 +1,81 @@
 #!/usr/bin/env python3
-
-from setuptools import setup
-from setuptools.command.install import install as orig_install
-
-from distutils.core import Command
+import os.path
+import stat
+import sys
 from distutils.command.build import build as orig_build
-
-from os import path
-
-from sys import platform
-
-from urllib.request import urlopen
-from http import HTTPStatus
-
+from distutils.core import Command
 from hashlib import sha512
+from http import HTTPStatus
+from urllib.request import urlopen
 
-from os import makedirs
-from os import fstat
-from os import fchmod
+from setuptools import setup
+from setuptools.command.install import install as orig_install
 
-from functools import lru_cache
+SHELLCHECK_VERSION = '0.7.0'
+PY_VERSION = '1'
 
 
 def get_arch() -> str:
-    if platform == "linux" or platform == "linux2":
+    if sys.platform.startswith('linux'):
         # TODO(rhee): detect "linux-aarch64" and "linux-armv6hf"
         return 'linux-x86_64'
-    elif platform == "darwin":
+    elif sys.platform == 'darwin':
         return 'darwin-x86_64'
-    elif platform == "win32" or platform == "win64":
+    elif sys.platform == 'win32':
         return 'exe'
     else:
-        raise RuntimeError('Unsupported platform')
+        raise ValueError('Unsupported platform')
 
 
 def get_executable_name() -> str:
-    return f'shellcheck-v{get_version()}.{get_arch()}'
-
-
-@lru_cache(maxsize=1)
-def get_version() -> str:
-    with open('VERSION', 'r') as fp:
-        return fp.read().strip()
+    return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return path.join(
-        'https://storage.googleapis.com/',
-        'shellcheck',
-        get_executable_name(),
-    )
+    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
 
 
 def download(url: str) -> bytes:
     with urlopen(url) as resp, urlopen(url + '.sha512sum') as shasum_resp:
-
         code = resp.getcode()
         scode = shasum_resp.getcode()
         if code != HTTPStatus.OK or scode != HTTPStatus.OK:
-            raise RuntimeError(f'HTTP failure. Codes: {code}, {scode}')
+            raise ValueError(f'HTTP failure. Codes: {code}, {scode}')
 
         data = resp.read()
         actual_shasum = sha512(data).hexdigest()
-        expected_shasum = shasum_resp.read().decode('utf8').split(" ")[0]
+        expected_shasum = shasum_resp.read().decode('utf8').split(' ')[0]
 
         if expected_shasum != actual_shasum:
-            raise RuntimeError(
-                'SHA512 mismatch! '
-                f'expected "{expected_shasum}" but got "{actual_shasum}"'
+            raise ValueError(
+                f'SHA512 mismatch! '
+                f'expected "{expected_shasum}" but got "{actual_shasum}"',
             )
 
         return data
 
 
 def save_executable(data: bytes, base_dir: str):
-    out_dir = path.join(base_dir, 'bin')
```

Comment:
windows is special snowflake and the directory is called `Scripts` there -- I corrected this below by using `install_scripts`